pkg/scheduler/schedulercache/node_info_test.go

```diff
@@ -0,0 +1,850 @@
+									Ports: []v1.ContainerPort{
+										{
+											HostIP:   "127.0.0.1",
+											HostPort: 80,
+											Protocol: "TCP",
+										},
+									},
+								},
+							},
+							NodeName: nodeName,
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Namespace: "node_info_cache_test",
+							Name:      "test-2",
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Resources: v1.ResourceRequirements{
+										Requests: v1.ResourceList{
+											v1.ResourceCPU:    resource.MustParse("200m"),
+											v1.ResourceMemory: resource.MustParse("1Ki"),
+										},
+									},
+									Ports: []v1.ContainerPort{
+										{
+											HostIP:   "127.0.0.1",
+											HostPort: 8080,
+											Protocol: "TCP",
+										},
+									},
+								},
+							},
+							NodeName: nodeName,
+						},
+					},
+				},
+			},
+			expected: &NodeInfo{
+				requestedResource:   &Resource{},
+				nonzeroRequest:      &Resource{},
+				allocatableResource: &Resource{},
+				generation:          2,
+				usedPorts: util.HostPortInfo{
+					"127.0.0.1": map[util.ProtocolPort]struct{}{
+						{Protocol: "TCP", Port: 80}:   {},
+						{Protocol: "TCP", Port: 8080}: {},
+					},
+				},
+				pods: []*v1.Pod{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Namespace: "node_info_cache_test",
+							Name:      "test-1",
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Resources: v1.ResourceRequirements{
+										Requests: v1.ResourceList{
+											v1.ResourceCPU:    resource.MustParse("100m"),
+											v1.ResourceMemory: resource.MustParse("500"),
+										},
+									},
+									Ports: []v1.ContainerPort{
+										{
+											HostIP:   "127.0.0.1",
+											HostPort: 80,
+											Protocol: "TCP",
+										},
+									},
+								},
+							},
+							NodeName: nodeName,
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Namespace: "node_info_cache_test",
+							Name:      "test-2",
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Resources: v1.ResourceRequirements{
+										Requests: v1.ResourceList{
+											v1.ResourceCPU:    resource.MustParse("200m"),
+											v1.ResourceMemory: resource.MustParse("1Ki"),
+										},
+									},
+									Ports: []v1.ContainerPort{
+										{
+											HostIP:   "127.0.0.1",
+											HostPort: 8080,
+											Protocol: "TCP",
+										},
+									},
+								},
+							},
+							NodeName: nodeName,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		ni := test.nodeInfo.Clone()
+		// Modify the field to check if the result is a clone of the origin one.
+		test.nodeInfo.generation += 10
+		if !reflect.DeepEqual(test.expected, ni) {
+			t.Errorf("expected: %#v, got: %#v", test.expected, ni)
+		}
+	}
+}
+
+func TestNodeInfoAddPod(t *testing.T) {
```

Comment:
This is not very different to TestNewNodeInfo as NewNodeInfo calls AddPod. May be you could just merge these.

Reply:
Or start with a node that already has some pods, and then add pod to the node.